core-bundle/contao/elements/ContentDownload.php

```diff
@@ -164,7 +164,9 @@ protected function getPreviews(string $path, string $downloadUrl): array
 
 		if ($this->fullsize)
 		{
-			if (!empty($GLOBALS['objPage']) && ($layoutId = $GLOBALS['objPage']->layout) && ($layout = LayoutModel::findById($layoutId)))
+			$pageModel = $container->get('contao.routing.page_finder')->getCurrentPage();
+
+			if ($pageModel && ($layoutId = $pageModel->layout) && ($layout = LayoutModel::findById($layoutId)))
```

Comment:
```suggestion
			if (($layoutId = $pageModel?->layout) && ($layout = LayoutModel::findById($layoutId)))
```

Nitpick: could be further shortened, just like in the other places.

Reply:
Then we could also shorten it to this I guess:

```php
if ($layout = LayoutModel::findById($pageModel?->layout ?? 0))
```

Reply:
It's probably better to not execute `Model::findById()` at all, if there is no ID available. I think our Model class would otherwise query unnecessarily for `id = 0`?